app/models/asset_field_type.rb

```diff
@@ -52,6 +53,11 @@ def mapping
 
   private
 
+  def asset_post_processing_data(metadata_hash)
+   return metadata_hash unless MimeMagic.new(field_data["asset"].content_type).mediatype == 'image'
```

Comment:
In more complex methods that you'd want to break out of early, I'm okay with multiple return paths, but this method could easily be re-written with a standard if/else